.github/workflows/build.yml

```diff
@@ -0,0 +1,249 @@
+
+            pytest -sv tests/integration
+          fi
+
+  # on a GitHub release, build the pip package and upload it to the appropriate pypi server.
+  # we upload prereleases to test.pypi.org and releases to pypi.org.
+  # the package is also uploaded to the GitHub release as an asset.
+  package-deploy:
+    needs: test
+
+    runs-on: ubuntu-18.04
+
+    if: github.event_name == 'release'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+
+      - name: set-release-env
+        shell: bash
+        run: |
+          RELEASE_TAG="${{ github.event.release.tag_name }}"
+          if [[ $RELEASE_TAG =~ ^v?([[:digit:]\.]+)(-rc)? ]]; then
+            VERSION="${BASH_REMATCH[1]}"
+            if [[ "${{ github.event.release.prerelease}}" == "true" ]]; then
+              # for staging builds we append the build number so we have
+              # distinct version numbers between prod and test pypi.
+              VERSION="$VERSION.$GITHUB_RUN_NUMBER"
+            fi
+
+          else
+            echo "Unable to parse deployment version from $RELEASE_TAG"
+            exit 1
+          fi
+
+          echo ::set-env name=VERSION::$VERSION
+
+      # ensure that the version file in the package will have the correct version
+      # matching the name of the tag
+      - name: update-version
+        shell: bash
+        run: |
+          if [[ -n "$VERSION" ]]; then
+            sed "s|\"latestVersion\":.*$|\"latestVersion\":\"$VERSION\",|g" synapseclient/synapsePythonClient > temp
+            rm synapseclient/synapsePythonClient
+            mv temp synapseclient/synapsePythonClient
+          fi
+
+      - id: build-package
+        shell: bash
+        run: |
+          python3 -m pip install setuptools==38.6.0
+
+          # install synapseclient
+          python3 setup.py install
+
+          # create distribution
+          python3 setup.py sdist
+
+          echo ::set-env name=PACKAGE_NAME::"synapseclient-${{env.VERSION}}.tar.gz"
+
+      # upload the artifact to the GitHub Action
+      - name: upload-build-artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.PACKAGE_NAME }}
+          path: dist/${{ env.PACKAGE_NAME }}
+
+      # upload the artifact as an asset of the GitHub Release
+      - name: upload-release-asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: https://uploads.github.com/repos/${{ github.event.repository.full_name }}/releases/${{ github.event.release.id }}/assets?name=${{ env.PACKAGE_NAME }}
+          asset_name: ${{ env.PACKAGE_NAME }}
+          asset_path: dist/${{ env.PACKAGE_NAME }}
+          asset_content_type: application/gzip
+
+      - name: deploy-to-pypi
+        shell: bash
+        run: |
+          python3 -m pip install twine==1.11.0
```

Comment:
version upgrade?

Reply:
The reason for the pinned twine was that it happens to have the same keyring dependency as synapseclient, so it could be installed into the same python environment that was used to build the client package itself.

I split the package and deploy into separate jobs, the deploy job now redownloads the complete artifact rather than building it so doesn't share any dependencies, allowing the specific version to be removed.